### PR TITLE
Commit chain metadata + scripts + builds adjustments

### PR DESCRIPTION
**Pull request by Lezek123**
Running `yarn update-chain-metadata` will update `chain-metadata.json` from current local node

Running `yarn verify-chain-metadata` will compare `chain-metadata.json` against the current local node metadata

**Build adjustments:**
- `hydra-typegen` will now use `chain-metadata.json` to generate the types, no need to connect to a running chain during query-node build step
- `@joystream/types` will now use `chain-metadata.json` to generate augmentations, no need to connect to a running chain during `generate:all` step
- most of the files generated by `@joystream/types generate:all` are now excluded from the repository
- `@joystream/types build` will now also run `@joystream/types generate:all`
- added pre-build cleanup in `@joystream/types build` via `git clean -xdf -e node_modules` in order to remove old build artifacts

### CODE REVIEW

**Comment by ahhda** [comment]
## Review and testing

### On Mac
* Query Node build runs successfully without the need to start a node
Screenshot 2021-10-18 at 10 52 21 AM

* Update chain metadata script works as expected
Screenshot 2021-10-18 at 10 53 40 AM

* Verify chain metadata script also works as expected
Screenshot 2021-10-18 at 10 55 05 AM

* Deleting and generating the metadata via scripts works as expected
Screenshot 2021-10-18 at 10 55 49 AM

### On Linux
* Query Node build runs successfully without the need to start a node
Screenshot 2021-10-18 at 11 09 19 AM

* Update and verify metadata scripts work as expected
Screenshot 2021-10-18 at 11 17 41 AM

Everything works as expected. We are good to go 🚀

File: apps.Dockerfile [hunk 1]
```diff
@@ -1,15 +1,6 @@
...
-# Change metadata.source in manifest file. It's not possible to override it via flag/env.
-USER root
-ARG WS_PROVIDER_ENDPOINT_URI
-COPY ./query-node/manifest.yml /joystream/qn-manifest.yml
-RUN yq e -i ".typegen.metadata.source = \"$WS_PROVIDER_ENDPOINT_URI\"" /joystream/qn-manifest.yml
-
 FROM --platform=linux/x86-64 node:14 as builder
 
 WORKDIR /joystream
-COPY . /joystream
```
Comment (ahhda) [inline apps.Dockerfile:11]: This will lead to no code being copied to the Container. We should keep this step.

Comment (Lezek123) [reply apps.Dockerfile:11]: Right, seems like I removed it by accident, thank you for spotting this! Fixed in https://github.com/Joystream/joystream/pull/2766/commits/6c22989c67cfc86dcd3a1203051a843d9b9a7595

**Review by ahhda — APPROVED** [review]

**Review by mnaamani — APPROVED** [review]

**Comment by mnaamani** [comment]
I just observed a minor issue when building docker images. The step
` git clean -xdf -e node_modules && yarn compile && yarn generate:all` fails because we do not copy the repo into the docker build context.

**Comment by mnaamani** [comment]
Maybe `git clean -xdf -e node_modules || yarn compile && yarn generate:all` is sufficient ?

**Pull request merged**